pkg/operator/revisioncontroller/revision_controller.go

```diff
@@ -191,20 +192,41 @@ func (c RevisionController) isLatestRevisionCurrent(revision int32) (bool, strin
 }
 
 func (c RevisionController) createNewRevision(recorder events.Recorder, revision int32) error {
+	// Before we create a new revision, check if any of the existing revisions got interrupted while InProgress
+	// If so, mark them Abandoned.
+	configMaps, err := c.configMapGetter.ConfigMaps(c.targetNamespace).List(context.TODO(), metav1.ListOptions{})
```

Comment:
do this *after* the new one is created and exclude the new one.  That way if you crash, you don't accidentally mark the last on as Abandoned before creating the new current one.